digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/LaneUtils.scala

```diff
@@ -0,0 +1,221 @@
+    val vkmRoadAddress = roadLinkTempDAO.getByRoadNumberRoadPartTrack(laneRoadAddressInfo.roadNumber.toInt, laneRoadAddressInfo.track, roadParts.toSet)
+
+    val mappedRoadLinks = roadLinkService.fetchVVHRoadlinks((roadAddresses.map(ra => ra.linkId) ++ vkmRoadAddress.map(_.linkId)).toSet)
+
+    val filteredLinkIds = mappedRoadLinks.filter { x =>
+      val roadPartNumber = x.attributes("ROADPARTNUMBER").toString.toInt
+      roadPartNumber >= laneRoadAddressInfo.initialRoadPartNumber || roadPartNumber <= laneRoadAddressInfo.endRoadPartNumber
+    }
+      .map(_.linkId)
+
+
+    val filteredRoadAddresses = roadAddresses.filter(x => filteredLinkIds.contains(x.linkId))
+
+    val lanesToInsert = newIncomeLanes.filter(_.id == 0)
+
+    val allLanesToCreate = filteredRoadAddresses.flatMap { road =>
+
+      val vvhTimeStamp = vvhClient.roadLinkData.createVVHTimeStamp()
+      val vvhRoadLink = mappedRoadLinks.find(_.linkId == road.linkId).get
+      val municipalityCode = vvhRoadLink.municipalityCode
+
+      lanesToInsert.map { lane =>
+
+        val laneCodeProperty = lane.attributes.properties.find(_.publicId == "lane_code").getOrElse(throw new IllegalArgumentException("Lane Code attribute not found!"))
+        val laneCode = laneCodeProperty.values.head.value.toString.toInt
+        val isMainLane = laneCode.toString.charAt(1).getNumericValue == 1
+
+        // Conversion from AddressMValue to MValue we used in LinkIds
+        val startDifferenceAddr = road.endAddrMValue - laneRoadAddressInfo.initialDistance
+        val startPoint = if(isMainLane) 0 else Math.abs(road.startMValue - (startDifferenceAddr * road.endMValue / road.endAddrMValue))
+        val endDifferenceAddr = road.endAddrMValue - laneRoadAddressInfo.endDistance
+        val endPoint = if(isMainLane) vvhRoadLink.length else Math.abs(road.startMValue - (endDifferenceAddr * road.endMValue / road.endAddrMValue))
+
+        if (road.roadPartNumber > laneRoadAddressInfo.initialRoadPartNumber && road.roadPartNumber < laneRoadAddressInfo.endRoadPartNumber) {
+
+          PersistedLane(0, road.linkId, sideCode, laneCode, municipalityCode,
+            road.startMValue, road.endMValue,
+            Some(username), Some(DateTime.now()),
+            None, None, expired = false,
+            vvhTimeStamp, None, lane.attributes)
+
+        }
+        else if (road.roadPartNumber == laneRoadAddressInfo.initialRoadPartNumber && road.roadPartNumber == laneRoadAddressInfo.endRoadPartNumber) {
+
+          if ( !(road.endAddrMValue > laneRoadAddressInfo.initialDistance && road.startAddrMValue < laneRoadAddressInfo.endDistance) )
+            None
+
+          else if (road.startAddrMValue <= laneRoadAddressInfo.initialDistance && road.endAddrMValue >= laneRoadAddressInfo.endDistance) {
+
+            PersistedLane(0, road.linkId, sideCode, laneCode, municipalityCode,
```

Comment:
all constructors only differ in this `startMeasure: Double, endMeasure: Double,` please create a method passing this two parameters

Reply:
Done